### ISSUE TITLE
508-defect-3 [AXE-CORE, SEMANTIC MARKUP]: id attribute value MUST be unique

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#13383]
# [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

Document has multiple static elements with the same ID attribute. All IDs **must** be unique (WCAG 4.1.1 (Level A)). In this case, the duplicated id is on both the container div and the h3, so this issue is relatively minor in impact. The screen reader will go to the first id on the div, and the h3 immediately follows so they won't be far off. It would be much clearer if the id was on the heading, so that the screen reader would then read the heading to verify arrival on the link anchor that the user selected to get here.

### Why it Matters

The ID attribute uniquely identifies elements on a page. It does not make sense to duplicate an ID.

Duplicate ID's can break the accessibility of labels for forms, table header cells, etc., by the second instance being skipped by screen readers, or by client-side scripts. They are common markup validation errors that can eliminate possible sources of accessibility problems, when not breaking accessibility.

Those experienced with client-side scripting know that when you re-use an id, typically the only one that gets acted upon by the scripting is the first instance of the use of that ID. Similarly, assistive technologies may, when referencing an id, only reference the first one accurately.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screen reader user, I want to select a link and receive verification that I arrived at the intended location.

## Environment

* Operating System: all
* Browser: all
* Server destination: staging

## Steps to Recreate

1. Enter https://staging.va.gov/manage-va-debt/your-debt/ in a browser, and please use Greg Anderson (user 1). This tests a user with a number of debts and debt letters.
2. Open Developer Tools and run an axe scan
3. Verify the "id attribute value must be unique" error appears, associated with `<div id="howDoIGetHelp" class="vads-u-font-family--sans">`

## Possible Fixes (optional)

Remove the id from the container div.

```diff
<!-- Current Code -->
! <div class="vads-u-font-family--sans" id="howDoIGetHelp">
!  <h3 class="vads-u-margin-top--4 vads-u-margin-bottom--0" id="howDoIGetHelp">How do I get financial help?</h3>
  <p class="vads-u-margin-top--2">If you need financial help, you can request:</p>

<!-- Recommended Fix -->
+ <div class="vads-u-font-family--sans">
+  <h3 class="vads-u-margin-top--4 vads-u-margin-bottom--0" id="howDoIGetHelp">How do I get financial help?</h3>
  <p class="vads-u-margin-top--2">If you need financial help, you can request:</p>

```

## WCAG or Vendor Guidance (optional)

* [Deque University - ID attribute values must be unique](https://dequeuniversity.com/rules/axe/3.5/duplicate-id?application=AxeChrome)

* [WCAG 4.1.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-parses.html): 

> 4.1.1 Parsing: In content implemented using markup languages, elements have complete start and end tags, elements are nested according to their specifications, elements do not contain duplicate attributes, and any IDs are unique, except where the specifications allow these features. (Level A)

## Screenshots or Trace Logs

![Screen Shot 2020-09-09 at 5 38 45 PM](https://user-images.githubusercontent.com/57469/92657786-67120480-f2c3-11ea-8e09-3816ca61898e.png)

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

**Comment by 1Copenut** [comment]
This should be breaking the CI/CD pipeline if the e2e tests have good coverage. If there's missing tests, that should be another issue, and this one could be bumped up to a higher priority.

**Comment by jenstrickland** [comment]
@leahkeeler While investigating this issue, would you please have the team review the CI/CD e2e tests, as @1Copenut highlights above?

**Comment by bradl3yC** [comment]
@jenstrickland please validate

**Comment by jenstrickland** [comment]
@bradl3yC 

Is there a new url I should use? I went to `https://staging.va.gov/manage-va-debt/your-debt/` and it 404'd. 

Also, was the fix to solely change the `id`s to make sure they are unique, or did it also address the e2e test issue that @1Copenut raised above? If the e2e tests include the axe checks in the CI/CD pipe this issue should have been caught in the dev process.

**Comment by jenstrickland** [comment]
Validated 🎉
